docs/admin/general-upgrade-guidelines.rst

```diff
@@ -0,0 +1,73 @@
+.. highlight:: sh
+
+.. _general_upgrade_guidelines:
+
+==========================
+General upgrade guidelines
+==========================
+
+Before kicking off an upgrade, there is a set of best practices to ensure the best outcome. Below you may find the five fundamental steps to prepare for an upgrade.
+
+.. NOTE::
+
+   This is not an exhaustive list, so you should consider your organization's specific needs and incorporate any additional steps or considerations that are relevant to your environment.
+
+Upgrade planning
+================
+
+Acknowledge breaking changes
+-----------------------------
+
+Review the `release notes`_ and documentation for the target version to understand any potential impact on existing functionality. 
+In case of a minor version upgrade, you should consider the intermediate versions. For example, when upgrading from 5.1 to 5.3, besides reviewing 5.3 release notes, the recommendation is to check for version 5.2 as well.
+
+Set up a test environment
+-------------------------
+
+Create a test environment that closely resembles your production environment, including the same CrateDB version, hardware, and network configuration. Populate the test environment with representative data and perform thorough testing to ensure compatibility and functionality, including functional and non-functional testing.
+
+
+Back up and plan recovery
+-------------------------
+
+Perform a cluster-wide backup of your production CrateDB and ensure you have a reliable recovery mechanism in place. Read more in the `snapshots`_ documentation. Once the backup is complete, validate it to ensure its usability.
```

Comment:
>  validate it to ensure its usability

Although it's a very valid point, as we figured it's not feasible to actually do so (no built-in validation feature/testing an actual restore takes too long), I guess we can skip that point here.